tests/rptest/tests/node_restart_probe_test.py

```diff
@@ -0,0 +1,282 @@
+
+    def __init__(self, **kvargs):
+        keys = kvargs.keys()
+        assert keys == self.KEYS, f"{keys=}, {self.KEYS=}"
+        dict.__init__(self, **{
+            k: self.build_value(v)
+            for k, v in kvargs.items()
+        })
+
+
+NO_RISKS = Risks(**{typ: set() for typ in Risks.KEYS})
+
+
+class NodeRestartProbeTest(RedpandaTest):
+    PRODUCE_BYTES = 300 * 1024 * 1024
+    MSG_SIZE = 1024
+
+    def __init__(self, test_context):
+        super(NodeRestartProbeTest, self).__init__(
+            test_context=test_context,
+            num_brokers=5,
+            extra_rp_conf={
+                'health_monitor_max_metadata_age': 100,  # ms
+                'enable_leader_balancer': False
+            })
+        self.admin = Admin(self.redpanda)
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+
+    def create_topics(self):
+        self.topics = [
+            TopicSpec(name='t1', partition_count=1, replication_factor=1),
+            TopicSpec(name='t3', partition_count=2, replication_factor=3),
+            TopicSpec(name='t5', partition_count=1, replication_factor=5),
+        ]
+        self.client().create_topic_with_assignment(self.topics[0].name, [[1]])
+        self.client().create_topic_with_assignment(self.topics[1].name,
+                                                   [[1, 2, 3], [3, 4, 5]])
+        self.client().create_topic(self.topics[2])
+
+    def get_node_risks(self, node) -> Risks:
+        reply = self.admin.get_broker_pre_restart_probe(node)
+        self.redpanda.logger.debug(f"get_risks returned: {reply}")
+        return Risks(**reply['risks'])
+
+    def get_risks(self) -> Dict[int, Risks]:
+        return {
+            self.redpanda.node_id(node): self.get_node_risks(node)
+            for node in self.redpanda.started_nodes()
+        }
+
+    def wait_pre_restart_probes(self,
+                                expected_risks: Dict[int, Risks],
+                                timeout_sec=30):
+        """wait until it returns expected result, make sure it 
+        does not return anything milder in the meanwhile"""
+        def risks_are_as_expected():
+            actual_risks = self.get_risks()
+            self.redpanda.logger.debug(
+                f"actual_risks={sorted(actual_risks.items())}, "
+                f"expected_risks={sorted(expected_risks.items())}")
+            return actual_risks == expected_risks
+
+        wait_until(risks_are_as_expected,
+                   timeout_sec=timeout_sec,
+                   backoff_sec=0.1,
+                   err_msg="Waiting for reported risks to match expected")
+
+    def produce_to_all_partitions(self, acks):
+        for topic in self.topics:
+            self.redpanda.logger.debug(f"producing to {topic.name}")
+            num_messages = int(topic.partition_count * self.PRODUCE_BYTES /
+                               self.MSG_SIZE)
+            self.kafka_tools.produce(topic.name, num_messages, self.MSG_SIZE,
```

Comment:
does this run nicely in debug? 300mb seems a bit too high

Reply:
Lower numbers make the test flaky for me. Yes I run it locally in debug.